src/GeoFormatTypes.jl

```diff
@@ -47,8 +57,9 @@ Base.show(io::IO, ::MIME"text/plain", ::Type{Geom}) = print(io, "Geometry mode")
 Trait specifying that a format object, like [`WellKnownText`](@ref),
 contains only coordinate reference system data.
 """
-struct CRS <: FormatMode end
-Base.show(io::IO, ::MIME"text/plain", ::Type{CRS}) = print(io, "CRS mode")
+struct CRS{T<:CRSMode} <: FormatMode end
+CRS() = CRS{UnknownCRS}()
+Base.show(io::IO, ::MIME"text/plain", ::Type{CRS{T}}) where {T} = print(io, "$T mode")
```

Comment:
AFAICT, this won't work for the CoordinateReferenceSystemFormat types?

Maybe we need either a wrapper as indicated in the issue, or an extra type param in the type definition for those

Reply:
Yeah, I guess they will all need a type parameter.

Reply:
Hmm @asinghvi17 your comment in the issue makes a good point that wrapping the other way is simpler, instead of adding heaps of type parameters to every object here we just compose the objects here as-is with other singletons.

Reply:
Can you show a MWE for a wrapper? I'd like to adjust the existing types to use them as they're implemented across the ecosystem.

Reply:
I was thinking of something on the lines of:
```julia
struct CRS{T, CRSTrait} <: CoordinateReferenceSystemFormat
	val::T 
end

CRS(x::CoordinateReferenceSystemFormat) = CRS(x, GI.crstrait(x))
```

Reply:
then, the `GI.crs` of a geometry might be `CRS{EPSG, GeographicCRSTrait}(EPSG((4326,)))`

Reply:
That MWE probably won't work as is, but you get the idea